src/cli/main.zig

```diff
@@ -2896,228 +3062,261 @@ fn rocBuild(allocs: *Allocators, args: cli_args.BuildArgs) !void {
 
-    std.fs.cwd().makePath(temp_dir) catch |err| switch (err) {
+    std.fs.cwd().makePath(build_cache_dir) catch |err| switch (err) {
         error.PathAlreadyExists => {},
         else => return err,
     };
 
-    // Generate app stub object file
-    const app_stub_obj = try app_stub.generateAppStubObject(allocs.arena, temp_dir, entrypoints, target);
-
-    // Get CRT files for the target
-    const crt_files = try target_mod.getVendoredCRTFiles(allocs.arena, target, platform_dir);
-
-    // Create object files list for linking
-    var object_files = try std.array_list.Managed([]const u8).initCapacity(allocs.arena, 16);
-
-    // Add our app stub and host library
-    try object_files.append(app_stub_obj);
-    try object_files.append(host_lib_path);
-
-    // Setup platform files based on target
-    var platform_files_pre = try std.array_list.Managed([]const u8).initCapacity(allocs.arena, 16);
-    var platform_files_post = try std.array_list.Managed([]const u8).initCapacity(allocs.arena, 16);
-    var extra_args = try std.array_list.Managed([]const u8).initCapacity(allocs.arena, 32);
-
-    // Add CRT files in correct order
-    if (crt_files.crt1_o) |crt1| try platform_files_pre.append(crt1);
-    if (crt_files.crti_o) |crti| try platform_files_pre.append(crti);
-    if (crt_files.crtn_o) |crtn| try platform_files_post.append(crtn);
-
-    // For static linking with musl, add libc.a
-    if (crt_files.libc_a) |libc| {
-        try platform_files_post.append(libc);
-    } else if (target.isDynamic()) {
-        // For dynamic linking with glibc, generate stub library for cross-compilation
-        // Check if we're doing actual cross-compilation
-        const is_cross_compiling = host_target != target;
-
-        if (is_cross_compiling) {
-            // For cross-compilation, use pre-built vendored stubs from the platform targets folder
-            const target_name = switch (target) {
-                .x64glibc => "x64glibc",
-                .arm64glibc => "arm64glibc",
+    // Get platform directory and host library (do this first to get platform source)
+    const app_dir = std.fs.path.dirname(args.path) orelse ".";
+    const platform_spec = extractPlatformSpecFromApp(allocs, args.path) catch |err| {
+        std.log.err("Failed to extract platform spec: {}", .{err});
+        return err;
+    };
+    std.log.debug("Platform spec: {s}", .{platform_spec});
+
+    // Resolve platform path
+    const platform_paths: ?PlatformPaths = if (std.mem.startsWith(u8, platform_spec, "./") or std.mem.startsWith(u8, platform_spec, "../"))
+        resolvePlatformSpecToPaths(allocs, platform_spec, app_dir) catch |err| blk: {
+            std.log.err("Failed to resolve platform paths: {}", .{err});
+            break :blk null;
+        }
+    else if (std.mem.startsWith(u8, platform_spec, "http://") or std.mem.startsWith(u8, platform_spec, "https://"))
+        resolveUrlPlatform(allocs, platform_spec) catch null
```

Comment:
We want to disallow `http` and only permit `https` -- let's address this in a follow up.

Reply:
Created #8605 to track